File: articles/scenario-modeling-for-complex-systems/fortran/scenario_score.f90
program scenario_score
  implicit none
  real :: v(4)
  real :: score

  v = (/1.55, 1.02, 1.28, 0.15/)
  score = 0.35*v(1) + 0.30*v(2) + 0.20*v(3) + 0.15*v(4)

  print *, "Robust resilience portfolio viability:", score
end program scenario_score

File: articles/scenario-modeling-for-complex-systems/cpp/scenario_score.cpp
#include <iostream>
#include <vector>

double viability(const std::vector<double>& v) {
    return 0.35 * v[0] + 0.30 * v[1] + 0.20 * v[2] + 0.15 * v[3];
}

int main() {
    std::vector<double> robust_resilience = {1.55, 1.02, 1.28, 0.15};
    std::cout << "Robust resilience portfolio viability="
              << viability(robust_resilience) << "\n";
    return 0;
}
